students/yooyoung/users/models.py

```diff
@@ -0,0 +1,23 @@
+from django.db import models
+from django.core.validators import RegexValidator
+
+
+class User(models.Model):
+    name = models.CharField(max_length=20)
+    email = models.EmailField(max_length=50, unique=True)
+    password = models.CharField(max_length=50)
+    note = models.TextField(max_length=100, null=True)
+    phone = models.CharField(
+        max_length=16,
+        blank=True,
+        null=True,
+        validators=[
+            RegexValidator(
+                regex=r'^\+?1?\d{9,15}$',
+                message="Phone number must be entered in the format '+123456789'. Up to 15 digits allowed."
+            ),
+        ],
+    )
```

Comment:
```suggestion
    phone = models.CharField(
        max_length=16,
        blank=True,
        null=True,
        validators=[
            RegexValidator(
                regex=r'^\+?1?\d{9,15}$',
                message="Phone number must be entered in the format '+123456789'. Up to 15 digits allowed."
            ),
        ],
    )
```
정규표현식 좋습니다~ 하지만 django 자체 validator를 사용해 models에서 처리하는 것은, form을 사용하지 않을 경우에는 의미가 없습니다. 저희는 template을 사용하지 않기 때문에 views에서 로직으로 정규표현식을 작성합니다!

Reply:
모델에서 디비 입력될 때 validation을 주는 것하고 view에서 post로 받을 때 validation을 주는 것의 차이가 궁금했는데 감사합니다~!